EIPS/eip-2767.md

```diff
@@ -0,0 +1,239 @@
+
+## Motivation
+
+<!--The motivation is critical for EIPs that want to change the Ethereum protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the EIP solves. EIP submissions without sufficient motivation may be rejected outright.-->
+
+Traditionally, many contracts that require that they be owned or controlled in some way use `ERC-173` which standardized the use of ownership in the smart contracts. For example to withdraw funds or perform administrative actions.
+
+```solidity
+contract dApp {
+  function doSomethingAdministrative() external onlyOwner {
+    // admin logic that can be performed by a single wallet
+  }
+}
+```
+
+Often, such administrative rights for a contract are written for maintenance purpose but users need to trust the owner. Rescue operations by an owner have raised questions on decentralised nature of the projects. Also, there is a possibility of compromise of an owner's private key.
+
+At present, there are various implementation ideas for handling governance. Adhering to a standard API would enable general tools (like EtherScan) to populate governance information, thus increasing transparency to users. This can result in a wide adoption for contract governance.
+
+## Specification
+
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Ethereum platforms (go-ethereum, parity, cpp-ethereum, ethereumj, ethereumjs, and [others](https://github.com/ethereum/wiki/wiki/Clients)).-->
+
+#### Notes
+
+- The following specifications use syntax from Solidity 0.6.0 (or above)
+- Governor is the board member who has voting right.
+- Power represents the voting privilege of a governor.
+
+A contract that is compliant with `ERC-2767` shall implement the following interface:
+
+```solidity
+/// @title ERC-2767 Governance with Privileged Voting Rights
+/// @dev ERC-165 InterfaceID: 0x4fe54581
+interface ERC2767 {
+    /// @dev This emits when governor power changes
+    event GovernorPowerUpdated(address indexed governor, uint256 power);
+
+    /// @notice Gets the consensus power of the governor
+    /// @param _governor Address of the governor
+    /// @return The governor's voting power
+    function powerOf(address _governor) external view returns (uint256);
+
+    /// @notice Gets the sum of the power of all governors
+    /// @return Sum of the power of all governors
+    function totalPower() external view returns (uint256);
+
+    /// @notice Gets number votes required for achieving consensus
+    /// @return Required number of consensus votes
+    function required() external view returns (uint256);
```

Comment:
Not all governed contracts will have voting.  Having voting be part of the interface may restrict its application.  A common (and IMO better) solution to governance is to have a single governor but that governor would be a multisig or something.  This means that the contract in question would *not* conform to this specification, and thus wouldn't show as having any governance.

Reply:
I think this EIP is more of a multi-sig interface. You are correct that this should not always have to be voting. Will improvise on this.

Reply:
@MicahZoltu I am not following.  How can you have decentralized governing without some kind of voting?   The IGovernanceOffchain interface in the standard provides a multisignature function for voting.